hldb/Lib/hldb.swift

```diff
@@ -102,196 +102,188 @@ public class HLDB {
-      case Text = "TEXT"
-      case Blob = "BLOB"
-      case Bool = "BOOL"
-    }
-    
-    public enum Index: String {
-      case None = "none"
-      case PrimaryKey = "primaryKey"
-      case Unique = "unique"
-      case Index = "index"
-      case Packed = "packed"
-      case Private = "private"
-    }
-    
-    public enum Default {
-      case None
-      case NonNull
-      case Value(AnyObject)
-    }
-    
-    public struct Field {
-      let name: String
-      let type: Type
-      let index: Index
-      let defaultValue: Default
-      
-      public init(name: String, type: Type, index: Index, defaultValue: Default) {
-        self.name = name
+    public init(fields: NSMutableDictionary) {
+      self.fields = fields
+    }
+  }
+
+public enum TableType: String {
+  case Integer = "INT"
+  case Real = "REAL"
+  case Text = "TEXT"
+  case Blob = "BLOB"
+  case Bool = "BOOL"
+}
+
+public enum TableIndex: String {
+  case None = "none"
+  case PrimaryKey = "primaryKey"
+  case Unique = "unique"
+  case Index = "index"
+  case Packed = "packed"
+  case Private = "private"
+}
+
+public enum TableDefault {
+  case None
+  case NonNull
+  case Value(AnyObject)
+}
+
+public struct TableField {
+  let name: String
+  let type: TableType
+  let index: TableIndex
+  let defaultValue: TableDefault
+  
+  public init(name: String, type: TableType, index: TableIndex, defaultValue: TableDefault) {
+    self.name = name
+    self.type = type
+    self.index = index
+    self.defaultValue = defaultValue
+  }
+  
+  public init(fromDict: NSDictionary) {
+    defaultValue = .NonNull
+    
+    if let name = fromDict["name"] as? String {
+      self.name = name
+    } else {
+      name = ""
+    }
+    
+    if let typeValue = fromDict["type"] as? String {
+      if let type = TableType(rawValue: typeValue) {
         self.type = type
-        self.index = index
-        self.defaultValue = defaultValue
-      }
-      
-      public init(fromDict: NSDictionary) {
-        defaultValue = .NonNull
-        
-        if let name = fromDict["name"] as? String {
-          self.name = name
-        } else {
-          name = ""
-        }
-        
-        if let typeValue = fromDict["type"] as? String {
-          if let type = Type(rawValue: typeValue) {
-            self.type = type
-          } else {
-            type = .Text
-          }
-        } else {
-          type = .Text
-        }
-        
-        if let isPrimaryKey = fromDict["pk"] as? Int {
-          if isPrimaryKey == 1 {
-            self.index = .PrimaryKey
-          } else {
-            index = .None
-          }
-        } else {
-          index = .None
-        }
-      }
-      
-      public func toDictionary() -> NSDictionary {
-        let outDict = NSMutableDictionary()
-        outDict["name"] = name
-        outDict["type"] = type.rawValue
-        outDict["index"] = index.rawValue
-        return outDict
+      } else {
+        type = .Text
```

Comment:
this can be `self.type = (fromDict["type"] as? String).flatmap{ TableType(rawValue: $0) } ?? .Text`